### PR TITLE
- add lock on writing, since this is mostly used on sqlite

### PR DESCRIPTION
**Pull request by kobimic**
and use the same loop for lock and server to allow the lock to work

### CODE REVIEW

**Review by tuxiqae — APPROVED** [review]
Looking good

File: jwthenticator/loop_management.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import asyncio
+from jwthenticator.consts import DB_URI
+
+loop = asyncio.new_event_loop()
+db_lock = asyncio.Lock(loop=loop)
+
+def is_using_sqlite()->bool:
+    return "sqlite" in DB_URI
```
Comment (GHadari) [inline jwthenticator/loop_management.py:8]: Add a newline please

Comment (GHadari) [inline jwthenticator/loop_management.py:8]: consider checking for ```sqlite://```

File: jwthenticator/tokens.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 from jwthenticator.schemas import JWTPayloadData, RefreshTokenData
 from jwthenticator.exceptions import InvalidTokenError, MissingJWTError
 from jwthenticator.consts import JWT_ALGORITHM, REFRESH_TOKEN_EXPIRY, JWT_LEASE_TIME, JWT_AUDIENCE, DB_URI
-
+from jwthenticator.loop_management import db_lock
```
Comment (GHadari) [inline jwthenticator/tokens.py:15]: Is this needed? its not mentioned in the rest of the file

File: jwthenticator/loop_management.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import asyncio
+from jwthenticator.consts import DB_URI
+
+loop = asyncio.new_event_loop()
```
Comment (GHadari) [inline jwthenticator/loop_management.py:4]: loop is very generic, could you please use a more specific name?

File: jwthenticator/tokens.py (hunk 2 above)
Comment (kobimic) [reply jwthenticator/tokens.py:15]: no, i will remove it

File: jwthenticator/loop_management.py (hunk 1 above)
Comment (kobimic) [reply jwthenticator/loop_management.py:8]: done

Comment (kobimic) [reply jwthenticator/loop_management.py:8]: done

**Review by GHadari — APPROVED** [review]

**Pull request merged**